src/huggingface_hub/utils/_headers.py

```diff
@@ -0,0 +1,139 @@
+            will be validated (cannot be `None`, cannot start by `"api_org***"`).
+
+    Returns:
+        A `Dict` of headers to pass in your API call.
+
+    Example:
+    ```py
+        >>> build_hf_headers(use_auth_token="hf_***") # explicit token
+        {"authorization": "Bearer hf_***"}
+
+        >>> build_hf_headers(use_auth_token=True) # explicitly use cached token
+        {"authorization": "Bearer hf_***"}
+
+        >>> build_hf_headers(use_auth_token=False) # explicitly don't use cached token
+        {}
+
+        >>> build_hf_headers() # implicit use of the cached token
+        {"authorization": "Bearer hf_***"}
+
+        # HF_HUB_DISABLE_IMPLICIT_TOKEN=True # to set as env variable
+        >>> build_hf_headers() # token is not sent
+        {}
+
+        >>> build_hf_headers(use_auth_token="api_org_***", is_write_action=True)
+        ValueError: You must use your personal account token for write-access methods.
+    ```
+
+    Raises:
+        [`ValueError`](https://docs.python.org/3/library/exceptions.html#ValueError)
+            If organization token is passed and "write" access is required.
+        [`ValueError`](https://docs.python.org/3/library/exceptions.html#ValueError)
+            If "write" access is required but token is not passed and not saved locally.
+        [`EnvironmentError`](https://docs.python.org/3/library/exceptions.html#EnvironmentError)
+            If `use_auth_token=True` but token is not saved locally.
+    """
+    # Get auth token to send
+    token_to_send = _get_token_to_send(use_auth_token)
+    _validate_token_to_send(token_to_send, is_write_action=is_write_action)
+
+    # Combine headers
+    headers = {}
+    if token_to_send is not None:
+        headers["authorization"] = f"Bearer {token_to_send}"
+    # TODO: add user agent in headers
+    return headers
+
+
+def _get_token_to_send(use_auth_token: Optional[Union[bool, str]]) -> Optional[str]:
+    """Select the token to send from either `use_auth_token` or the cache."""
+    # Case token is explicitly provided
+    if isinstance(use_auth_token, str):
+        return use_auth_token
+
+    # Case token is explicitly forbidden
+    if use_auth_token is False:
+        return None
+
+    # Token is not provided: we get it from local cache
+    cached_token = HfFolder().get_token()
+
+    # Case token is explicitly required
+    if use_auth_token is True:
+        if cached_token is None:
+            raise EnvironmentError(
+                "Token is required (`use_auth_token=True`), but no token found. You"
+                " need to provide a token or be logged in to Hugging Face with"
+                " `huggingface-cli login` or `notebook_login`. See"
+                " https://huggingface.co/settings/tokens."
+            )
+        return cached_token
+
+    # Case implicit use of the token is forbidden by env variable
+    if os.environ.get("HF_HUB_DISABLE_IMPLICIT_TOKEN"):
```

Comment:
(nit) In `datasets` and `transformers`, we usually check if the value is not in
```python
ENV_VARS_TRUE_VALUES = {"1", "ON", "YES", "TRUE"}
```

Reply:
Addressed in https://github.com/huggingface/huggingface_hub/pull/1081